@LinProp/LinProp.m

```diff
@@ -431,161 +431,347 @@ function display(obj)
-                        end
-                        B = LinProp.Convert2LinProp(bm);
-                        s = size(B);
-                        if length(s) > 2 && s(end) == 1
-                            B = reshape(B, s(1:end-1));
-                        end
+        %SUBSREF Subscripted reference.
+        %   A(I) is an array formed from the elements of A specified by the
+        %   subscript vector I.  The resulting array is the same size as I except
+        %   for the special case where A and I are both vectors.  In this case,
+        %   A(I) has the same number of elements as I but has the orientation of A.
+        %
+        %   A(I,J) is an array formed from the elements of the rectangular
+        %   submatrix of A specified by the subscript vectors I and J.  The
+        %   resulting array has LENGTH(I) rows and LENGTH(J) columns.  A colon used
+        %   as a subscript, as in A(I,:), indicates all columns of those rows
+        %   indicated by vector I. Similarly, A(:,J) = B means all rows of columns
+        %   J.
+        %
+        %   For multi-dimensional arrays, A(I,J,K,...) is the subarray specified by
+        %   the subscripts.  The result is LENGTH(I)-by-LENGTH(J)-by-LENGTH(K)-...
+        
+            if strcmp('.', {S.type})
+                error('Dot indexing is not supported for variables of this type.');   %TODO: This was allowed in  V2.4.7 of the uncLib MATLAB wrapper - but did not serve any function, as far as Dion can see.
+            elseif strcmp('{}', {S.type})
+                error('Brace indexing is not supported for variables of this type.');
+            elseif length(S) > 1
+                error('Invalid array indexing.');    % This type of error should never appear, as it would require multiple round brackets.
+            end
+            
+            ni = numel(S.subs);
+            if ni == 0
+                B = A;
+                return;
+            elseif ni == 1 && isempty(S.subs{1})
+                B = LinProp([]);
+                return;
+            end
+            
+            sizeA = size(A);
+            isvectorA = numel(sizeA) == 2 && any(sizeA == 1);
+            src_subs = S.subs;
+            transpose_vector = false;
+            output_shape = [];
+
+            % This is a very special case. If linear indexing is used, but
+            % the linear indexes are arranged in form of a vector, 
+            if ni == 1 && ~isvector(src_subs{1})
+                if islogical(src_subs{1})
+                    src_subs{1} = src_subs{1}(:);
+                else
+                    output_shape = size(src_subs{1});   % Save shape of output for later.
+                    src_subs{1} = src_subs{1}(:);       % But conform to vector for processing.
+                end
+            end
+            
+            % TODO: We could probably simplify this code if we would used 
+            % SetItemsNd instead of SetItems1d for 1-by-N vectors.
```

Comment:
SetItemsNd needs to compute the flatindex inside METAS UncLib. SetItems1d doesn't compute the flatindex.
The overhead of using SetItemsNd is not too much. Only integer multiplication and an additional array initialization.

So I think we should simplify the MATLAB wrapper.

Reply:
This idea proved to be more trouble than it was worth. Instead, I simplified the code be reodering code segements and replacing all uses of the variable transpose_vector with the variable output_shape.